hibernate-core/src/main/java/org/hibernate/metamodel/model/domain/internal/JpaMetamodelImpl.java

```diff
@@ -490,21 +495,54 @@ private void applyNamedEntityGraphs(Collection<NamedEntityGraphDefinition> named
 			if ( entityType == null ) {
 				throw new IllegalArgumentException(
 						"Attempted to register named entity graph [" + definition.getRegisteredName()
-								+ "] for unknown entity [" + definition.getEntityName() + "]"
+						+ "] for unknown entity [" + definition.getEntityName() + "]"
 
 				);
 			}
+
 			final NamedEntityGraph namedEntityGraph = definition.getAnnotation();
 			final RootGraphImpl<?> entityGraph =
-					createRootGraph( definition.getRegisteredName(), entityType,
-							namedEntityGraph.includeAllAttributes() );
-			if ( namedEntityGraph.attributeNodes() != null ) {
-				applyNamedAttributeNodes( namedEntityGraph.attributeNodes(), namedEntityGraph, entityGraph );
-			}
+					createEntityGraph(
+							namedEntityGraph,
+							definition.getRegisteredName(),
+							entityType,
+							namedEntityGraph.includeAllAttributes()
+					);
+
 			entityGraphMap.put( definition.getRegisteredName(), entityGraph );
 		}
 	}
 
+	private <T> RootGraphImpl<T> createEntityGraph(
+			NamedEntityGraph namedEntityGraph,
+			String registeredName,
+			EntityDomainType<T> entityType,
+			boolean includeAllAttributes) {
+		final RootGraphImpl<T> entityGraph =
+				createRootGraph( registeredName, entityType, includeAllAttributes );
+
+		if ( namedEntityGraph.subclassSubgraphs() != null ) {
+			Arrays
+					.stream( namedEntityGraph.subclassSubgraphs() )
+					.forEach( subclassSubgraph -> {
+						GraphImplementor<?> subgraph = (GraphImplementor<?>) entityGraph.addTreatedSubgraph(
+								(Class) subclassSubgraph.type() );
+
+						applyNamedAttributeNodes(
+								subclassSubgraph.attributeNodes(),
+								namedEntityGraph,
+								subgraph
+						);
+					} );
```

Comment:
Is there a particular reason to use `Arrays.stream().forEach()` here instead of just `for`?

Reply:
Not really i can change if you want

Reply:
modified to maintain consistency